src/components/views/rooms/wysiwyg_composer/hooks/useSuggestion.ts

```diff
@@ -153,18 +155,34 @@ export function processMention(
 
     const { node } = suggestionData;
 
-    const textBeforeReplacement = node.textContent?.slice(0, suggestionData.startOffset) ?? "";
-    const textAfterReplacement = node.textContent?.slice(suggestionData.endOffset) ?? "";
+    // create an <a> element with the required attributes to allow us to interpret the mention as being a pill
+    const linkElement = document.createElement("a");
+    const linkTextNode = document.createTextNode(displayName);
+    linkElement.setAttribute("href", href);
+    linkElement.setAttribute("contenteditable", "false");
+    Object.entries(attributes).forEach(
+        ([attr, value]) => isNotUndefined(value) && linkElement.setAttribute(attr, value),
+    );
+    linkElement.appendChild(linkTextNode);
+
+    // create text nodes to go before and after the link
+    const leadingTextNode = document.createTextNode(node.textContent?.slice(0, suggestionData.startOffset) || "\u200b");
+    const trailingTextNode = document.createTextNode(` ${node.textContent?.slice(suggestionData.endOffset) ?? ""}`);
```

Comment:
I guess it's not a problem if either of these is empty?

Reply:
Yep, that's right